src/DynamoCore/Graph/Nodes/PortModel.cs

```diff
@@ -369,6 +369,23 @@ protected override void DeserializeCore(XmlElement nodeElement, SaveContext cont
         }
 
         #endregion
+
+        public bool Equals(PortModel other)
```

Comment:
last time I tried overriding `equals` there were a bunch of unexpected failures where reference equality was expected throughout the codebase instead of equality by guid - (I can't remember what type this was on) - just a consideration.

Reply:
In this case, I'm checking for guid equality only if reference equality fails. From what I see in other nodemodel types, it looks like the equality check will either be true or false in both cases (reference equality + guid equality) since if the same port model is being reused, both conditions will be true, and if a new port model is created, still both conditions will be false, unless the new ports are created using the same guids as the old ones, which is what I'm enforcing here explicitly for dummy nodes.